src/benchmarking/directory_parser.py

```diff
@@ -0,0 +1,207 @@
+import os
+import sys
+import argparse
+import csv
+from pathlib import Path
+from enum import Enum
+from dynaconf import Dynaconf
+
+
+class ToolPathFinder(Enum):
+    # ENRICHSEQ = 'enrichseq/merge_overlap_filter/merge_overlap_out_refined.csv'
+    ENRICHSEQ = 'enrichseq/output_files/taxid_abundances.csv'
+    FASTVIROMEEXPLORER = 'FastViromeExplorer-final-sorted-abundance.tsv'
+    BRACKEN = 'abundances.bracken'
+    BRACKEN_ASSEMBLED = 'bracken/assembled_abundances.bracken'
+    BRACKEN_NO_ASSEMBLY = 'bracken/nonassembled_abundances.bracken'
+
+
+def parse_directory_trees(truth_path: str, result_path: str, output_prefix: str) -> None:
+    '''
+    DESCRIPTION:
+        Get directory structure metadata from file paths and write to CSV.
+        Handles truth and result directory trees
+    
+    INPUT:
+        1. Path to truth directory tree
+        2. Path to results directory tree (i.e. the results of the tools that were tested) 
+        3. Path to desired place to store CSV file
+
+    OUTPUT:
+        CSV file containing: <trial #>,<tool>,<test>,<subtest>,<path to final result file>
+        e.g. 3,EnrichSeq,num_genomes,10_genomes_500000_reads,/projects/benchmarking/results/test_3/EnrichSeq/num_genomes/10_genomes_500000_reads/enrichseq/output_files/taxid_abundances.csv
+    '''
+    
+    # create CSV file to append to
+    output_csv_path = Path(settings.OUTPUT_DIRECTORY) / Path(output_prefix + '.csv')
+
+    # if os.path.exists(output_csv_path):
+    #     os.remove(output_csv_path)
+
+    with open(output_csv_path, 'w') as csvfile:
+        writer = csv.writer(csvfile) 
+        writer.writerow(['Trial_Num', 'Experiment', 'Condition', 'Tool', 'File_Path'])
+
+    # parse truth
+    truth_rows = parse_truth_filepaths(truth_path)
+    file = write_to_csv(truth_rows, output_csv_path)
+
+    # parse results
+    result_rows = parse_results_filepaths(result_path)
+    file = write_to_csv(result_rows, output_csv_path)
+    
+    print(f'Metadata saved to:{file}')
```

Comment:
The [logging std lib](https://docs.python.org/3/howto/logging.html) could be used here.